jwst/pathloss/pathloss.py

```diff
@@ -752,19 +832,41 @@ def _corrections_for_mos(slit, pathloss, exp_type, source_type=None):
         # Calculate the 1-d wavelength and pathloss vectors
         # for the source position
         # Get the aperture from the reference file that matches the slit
-        nshutters = util.get_num_msa_open_shutters(slit.shutter_state)
-        aperture = get_aperture_from_model(pathloss, nshutters)
+        slitlength = len(slit.shutter_state)
+        aperture = get_aperture_from_model(pathloss, slit.shutter_state)
+        log.info(f"Shutter state = {slit.shutter_state}, using {aperture.name} entry in ref file")
+        two_shutters = False
+        if shutter_below_is_closed(slit.shutter_state) and not shutter_above_is_closed(slit.shutter_state):
+            ycenter = ycenter - 1.0
+            log.info('Shutter below fiducial is closed, using lower region of pathloss array')
+            two_shutters = True
```

Comment:
I may be misreading this but it seems like if there's a long slitlet, as in `shutter_state == 'x1111'`, it will fall into this case. Is the correction computed over all shutters or is it computing it for the two_shutter case only?

Reply:
Yes, my understanding is that the correction depends on the location of the target in the slit, and this location is used to get a 1-d pathloss correction vs wavelength.  This correction is applied to all pixels in the data, with the correction interpolated as a function of wavelength only.  In other words, all points in the corrected data with the same wavelength will have the same correction, regardless of which shutter the pixel is in.